### ISSUE TITLE
[naming-convention] Add a selector for interface members only

### ISSUE THREAD

**Issue by asdf93074** [typescript-eslint/typescript-eslint#4218]
We have a camelCase convention in our codebase for defining variables for both classes and interfaces. Unfortunately, some of our interfaces which map directly to server responses and requests require snake_case conventions for their variables. We would like it if we could specifically control what convention should be followed for members of an interface.

```JSON
{
  "rules": {
    "@typescript-eslint/naming-convention": [
      "error",
      {
        "selectors": ["interfaceMember"],
        "format": ["PascalCase", "snake_case"]
      }
    ]
  }
}
```

```TS
interface ICorrectServerResponse {
  id: number;
  first_name: string;
  last_name: string;
}

interface IBadServerResponse {
  Id: number;
  First_Name: string;
  lastName: string;
}
```

**Expected Result**

Passes for the ICorrectServerResponse interface while prints an error for the IBadServerResponse message.

**Additional Info**

We already have an 'enumMember' selector so I daresay this would just drive towards better member coverage through selectors.

**Comment by bradzacher** [comment]
`typeProperty` is the selector you want - it will match object literal type properties and interface properties.

Enum members being separated from other things makes sense as it's very common for people to name enum members upper case or Pascal case (contrary to camel or snake casing for everything else).

I don't think splitting object literal type properties from interface properties from object literal type properties is necessarily a good idea because the two are completely interchangeable.

**Comment by asdf93074** [comment]
> typeProperty is the selector you want - it will match object literal type properties and interface properties.

Cool, it works as intended but now, my object literal properties have a different format which seems to not pass the naming-convention rule. Is there a way to turn formatting off for object literal properties? I tried setting the format as null with the `objectLiteralProperty` selector but it still seems to fail under the format for `typeProperty` and `variable` selectors.
